framework/db/src/com/cloud/utils/db/GenericDaoBase.java

```diff
@@ -969,7 +969,12 @@ public T findByUuidIncludingRemoved(final String uuid) {
     @Override
     @DB()
     public T findByIdIncludingRemoved(ID id) {
-        return findById(id, true, null);
+        if (_cache != null) {
+            final Element element = _cache.get(id);
+            return element == null ? findById(id, true, null) : (T)element.getObjectValue();
+        } else {
```

Comment:
yes, but I think the submitted code suggest it is optional:
```
T result = findById(id, true, null)
if (_cache != null) {
    final Element element = _cache.get(id);
    if (element != null) {
        result = (T)element.getObjectValue();
    }
}
return result;
```
I have no problem with the proposed code if _cache is optional however.